docs/adr/001_error_handling.md

```diff
@@ -0,0 +1,107 @@
+
+There are various ways to handle errors on trait methods, including swallowing them and logging, panicing, returning a shared global error, or returning a method-specific error. We strive for consistency, and we want to be sure that we've put enough thought into what this looks like that we don't have to make breaking interface changes unecessarily in the future.
+
+This was discussed extensively in [#2571](https://github.com/open-telemetry/opentelemetry-rust/issues/2571).
+
+
+## Related Work
+
+* [#2564](https://github.com/open-telemetry/opentelemetry-rust/issues/2564)
+* [#2561](https://github.com/open-telemetry/opentelemetry-rust/issues/2561)
+* [#2381](https://github.com/open-telemetry/opentelemetry-rust/issues/2381)
+
+## Considered Options
+
+**Option 1: Continue as is**
+Continue the status quo, returning a mix of either nothing or the trait-wide error type. This is inconsistent and limits the caller's ability to handle errors. 
+
+**Option 2: Extend trait-wide error type to all methods on trait**
+In this option we have an error type per trait regardless of the potential error paths for the individual methods. Concretely if `fn (a)` can return `Failure1` and `Failure2`,  and `fn (b)` can return `FailureC`, we have a unified error type that contains `Failure`, `Failure2`, and `Failure3`.
+
+ This will mean that callers will have to know how and when to discard errors from a particular API call based on an understanding of which subset of errors that particular call can make. 
+
+Conversely, it will reduce the number of error types in the code base. 
+
+**Option 3: Use shared errors where practical across signals, devolve into individual errors per operation if they need to diverge**
+
+Here we aim to consolidate error types where possible _without indicating a function may return more errors than it can actually return_. Conversely in **Option 2**, a caller of either of the example functions is forced to handle or discard all errors. In this case, we choose to sacrifice the single error and diverge into a separate error for each operation.
+
+Our preference for error types is thus:
+
+1. Consolidated error that covers all methods of a particular "trait type" (e.g., signal export) and method
+1. Devolves into error type per method of a particular trait type (e.g., `SdkShutdownResult`, `SdkExportResult`) _if the error types need to diverge_
+1. May alternatively devolve into error type per signal (e.g., `SpanExporter`) if the _signals diverge_
+
+This approach generalises across both **signals** and **trait methods**. For example, returning to our exporter traits, we have a trait that looks the same for each signal, with the same three methods. Upon closer inspection ([#2600](https://github.com/open-telemetry/opentelemetry-rust/issues/2600)), the potential error set is the same both between the methods *and* between the signals; this means we can use a single shared error type across both axes:
+
+```rust
```

Comment:
let us put this under a header - Example/something, so its easy to link to from other discussions.